test/e2e/spacebindingrequest_migration_test.go

```diff
@@ -0,0 +1,143 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	spacebindingrequesttestcommon "github.com/codeready-toolchain/toolchain-common/pkg/test/spacebindingrequest"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	testsupportspace "github.com/codeready-toolchain/toolchain-e2e/testsupport/space"
+	testsupportspacebinding "github.com/codeready-toolchain/toolchain-e2e/testsupport/spacebinding"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestMigrateSpaceBindingToSBR(t *testing.T) {
+	// given
+	awaitilities := WaitForDeployments(t)
+	// member cluster configured to skip user creation to mimic stonesoup configuration where user & identity resources are not created
+	memberConfigurationWithSkipUserCreation := testconfig.ModifyMemberOperatorConfigObj(awaitilities.Member1().GetMemberOperatorConfig(t), testconfig.SkipUserCreation(true))
+	// configure default space tier to appstudio
+	awaitilities.Host().UpdateToolchainConfig(t, testconfig.Tiers().DefaultUserTier("deactivate30").DefaultSpaceTier("appstudio"), testconfig.Members().Default(memberConfigurationWithSkipUserCreation.Spec))
+	// setup some users and spaces on m1
+	primaryMUR, primarySpace := createUser(t, awaitilities, "space-owner", awaitilities.Member1()) // we provision one space on m1 that will be shared with other users in the same cluster
+	guestMUR, _ := createUser(t, awaitilities, "space-guest", awaitilities.Member1())
+	spaceGuestMURwithSBR, _ := createUser(t, awaitilities, "space-guest-sbr", awaitilities.Member1())
+	// setup some users and spaces on m2
+	primaryMUR1, primarySpace1 := createUser(t, awaitilities, "space-owner1", awaitilities.Member2()) // we provision one space on m2 that will be shared with other users in the same cluster
+	guestMUR1, _ := createUser(t, awaitilities, "space-guest1", awaitilities.Member2())
+	guestMUR2, _ := createUser(t, awaitilities, "space-guest2", awaitilities.Member2())
+
+	// when
+	// we add the spaceGuests to the primarySpace using a spacebinding
+	guestSpaceBinding := testsupportspacebinding.CreateSpaceBinding(t, awaitilities.Host(), guestMUR, primarySpace, "admin")
+	tier, err := awaitilities.Host().WaitForNSTemplateTier(t, primarySpace.Spec.TierName)
+	require.NoError(t, err)
+	_, err = awaitilities.Member1().WaitForNSTmplSet(t, primarySpace.Name,
+		wait.UntilNSTemplateSetHasSpaceRoles(
+			wait.SpaceRole(tier.Spec.SpaceRoles["admin"].TemplateRef, primaryMUR.Name, guestMUR.Name)))
+	require.NoError(t, err)
```

Comment:
I means really down - after we verify the expected changes (migration) of the SpaceBindings and SpaceBindingRequest CRs. 
The flow of the verification I had in my mind was:
1. verify that the expected SB were replaced by the new one manage via SBR
2. verify that the default SB is still present
3. verify that the NSTemplateSet is in the expected format - none of the previous steps broke the list of Roles and usernames

Reply:
Ok I see, I wanted the test to "fail quickly" in case something is not ok, since in the end what matters is the NSTemplateSet. But also this might make the failure a bit cryptic , since it could be because of many reasons. Makes sense to have it as last check.

I've shuffled things a bit, could you please take another look at https://github.com/codeready-toolchain/toolchain-e2e/pull/834/commits/1d0004d6569b59c455427af6f9364f548aa84f53

The assertions are still grouped by member cluster, but hopefully this is more close to what your were thinking about.